pika/connection.py

```diff
@@ -612,9 +614,12 @@ def __init__(self,  # pylint: disable=R0913,R0914,R0912
         :param pika.credentials.Credentials credentials: auth credentials
         :param int channel_max: Maximum number of channels to allow
         :param int frame_max: The maximum byte size for an AMQP frame
-        :param int heartbeat: Heartbeat timeout. Max between this value
-            and server's proposal will be used as the heartbeat timeout. Use 0
-            to deactivate heartbeats and None to accept server's proposal.
+        :param int|callable heartbeat: Heartbeat timeout. If set and is
+            an integer value, it will be used during connection tuning.
+            If set and is a callable, it will be invoked during connection 
+            tuning and passed the server's proposal. Its return value will 
```

Comment:
I don't understand the meaning of "passed the server's proposal". I think you intended to say "its return value will be used to override the server's proposal".

Reply:
I meant passed the server's proposal as an argument to the callback. I will change the wording ;)